lib/node/workspace.dart

```diff
@@ -25,13 +25,193 @@ class ViewRegistry extends ProxyHolder {
     _instance = this;
   }
 
-  // TODO: expose addViewProvider(providerSpec)
+  // TODO: add modelConstructor
+  Disposable addViewProvider(createView) =>
+      invoke('addViewProvider', createView);
 
   /// Get the view associated with an object in the workspace. The result is
   /// likely an html Element.
   dynamic getView(object) => invoke('getView', object);
 }
 
+// A Change Listener for events going into Atom instead of out.
+class JsChangeListener {
+  List<JsFunction> _callbacks = [];
+
+  void add(JsFunction callback) {
+    _callbacks.add(callback);
+    return jsify ({
+      'dispose': () => _callbacks.remove(callback)
+    });
+  }
+
+  void change(List parameters) {
+    for (var callback in _callbacks) {
+      callback?.apply(parameters);
+    }
+  }
+}
+
+/// The term "item" refers to anything that can be displayed
+/// in a pane within the workspace, either in the {WorkspaceCenter} or in one
+/// of the three {Dock}s. The workspace expects items to conform to the
+/// following interface:
+class Item extends ProxyHolder {
+
+  JsChangeListener onDidChangeTitle = new JsChangeListener();
+
+  String _title;
+
+  Item(JsObject object) : super(object);
+
+  Item.fromFields({element, title, uri, defaultLocation, destroy})
```

Comment:
Can any of these be reasonably types (`String`)? If no, perhaps an explicit `dynamic`?